x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/Join.java

```diff
@@ -106,31 +101,49 @@ public List<Attribute> output() {
     }
 
     /**
-     * Merge output fields.
-     * Currently only implemented for LEFT JOINs; the rightOutput shadows the leftOutput, except for any attributes that
-     * occur in the join's matchFields.
+     * Combine the two lists of attributes into one.
+     * In case of (name) conflicts, specify which sides wins, that is overrides the other column - the left or the right.
      */
     public static List<Attribute> computeOutput(List<Attribute> leftOutput, List<Attribute> rightOutput, JoinConfig config) {
-        AttributeSet matchFieldSet = new AttributeSet(config.matchFields());
-        Set<String> matchFieldNames = new HashSet<>(Expressions.names(config.matchFields()));
-        return switch (config.type()) {
-            case LEFT -> {
-                // Right side becomes nullable.
-                List<Attribute> fieldsAddedFromRight = removeCollisionsWithMatchFields(rightOutput, matchFieldSet, matchFieldNames);
-                yield mergeOutputAttributes(fieldsAddedFromRight, leftOutput);
-            }
-            default -> throw new UnsupportedOperationException("Other JOINs than LEFT not supported");
-        };
+        JoinType joinType = config.type();
+        List<Attribute> output;
+        if (LEFT.equals(joinType)) {
+            // right side becomes nullable and overrides left
+            // output = merge(leftOutput, makeNullable(rightOutput));
+            output = merge(leftOutput, rightOutput);
+        } else if (RIGHT.equals(joinType)) {
+            // left side becomes nullable and overrides right
+            // output = merge(makeNullable(leftOutput), rightOutput);
+            output = merge(leftOutput, rightOutput);
+        } else {
+            throw new IllegalArgumentException(joinType.joinName() + " unsupported");
+        }
+        return output;
     }
 
-    private static List<Attribute> removeCollisionsWithMatchFields(
-        List<Attribute> attributes,
-        AttributeSet matchFields,
-        Set<String> matchFieldNames
-    ) {
-        List<Attribute> result = new ArrayList<>();
+    /**
+     * Merge the two lists of attributes into one and preserves order.
+     * In case of conflicts, specify the existing entry is overridden (in place) or not.
```

Comment:
Doc doesn't match implementation (yet?). Currently the right overrides the left, no way to specify otherwise.